plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/machine/backup/WorkspaceFsBackupScheduler.java

```diff
@@ -43,112 +41,98 @@
-    private final WorkspaceRuntimes             workspaceRuntimes;
-    private final MachineBackupManager          backupManager;
-    private final Map<String, Long>             lastMachineSynchronizationTime;
-    private final ExecutorService               executor;
-    private final ConcurrentMap<String, String> devMachinesBackupsInProgress;
-    private final Integer                       syncPort;
+    private final long                                  syncTimeoutMillisecond;
+    private final WorkspaceRuntimes                     workspaceRuntimes;
+    private final Map<String, Long>                     lastWorkspaceSynchronizationTime;
+    private final ExecutorService                       executor;
+    private final ConcurrentMap<String, String>         workspacesBackupsInProgress;
+    private final Map<String, EnvironmentBackupManager> backupManagers;
+    private final WorkspaceManager                      workspaceManager;
 
     @Inject
-    public WorkspaceFsBackupScheduler(WorkspaceRuntimes workspaceRuntimes,
-                                      MachineBackupManager backupManager,
+    public WorkspaceFsBackupScheduler(Map<String, EnvironmentBackupManager> backupManagers,
+                                      WorkspaceRuntimes workspaceRuntimes,
                                       @Named("machine.backup.backup_period_second") long syncTimeoutSecond,
-                                      @Named("codenvy.workspace.projects_sync_port") Integer syncPort) {
+                                      WorkspaceManager workspaceManager) {
         this.workspaceRuntimes = workspaceRuntimes;
-        this.backupManager = backupManager;
+        this.backupManagers = backupManagers;
         this.syncTimeoutMillisecond = TimeUnit.SECONDS.toMillis(syncTimeoutSecond);
-        this.syncPort = syncPort == null || syncPort == 0 ? null : syncPort;
+        this.workspaceManager = workspaceManager;
 
         this.executor = Executors.newCachedThreadPool(
-                new ThreadFactoryBuilder().setNameFormat("MachineFsBackupScheduler-%s")
+                new ThreadFactoryBuilder().setNameFormat("WorkspaceFsBackupScheduler-%s")
                                           .setUncaughtExceptionHandler(LoggingUncaughtExceptionHandler.getInstance())
                                           .build());
-        this.lastMachineSynchronizationTime = new ConcurrentHashMap<>();
-        this.devMachinesBackupsInProgress = new ConcurrentHashMap<>();
+        this.lastWorkspaceSynchronizationTime = new ConcurrentHashMap<>();
+        this.workspacesBackupsInProgress = new ConcurrentHashMap<>();
     }
 
     @ScheduleRate(initialDelay = 1, period = 1, unit = TimeUnit.MINUTES)
     public void scheduleBackup() {
         for (Map.Entry<String, WorkspaceRuntimes.WorkspaceState> wsStateEntry :
                 workspaceRuntimes.getWorkspaces().entrySet()) {
 
+            String workspaceId = wsStateEntry.getKey();
```

Comment:
I don't see usage of `wsStateEntry.getValue()`. Would it better to use `workspaceRuntimes.getWorkspaces().keySet()` for loop?